core/chainio/avs_writer.go

```diff
@@ -113,7 +113,16 @@ func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMe
 		if err != nil {
 			return nil, err
 		}
-		previousTxGasPrice := txOpts.GasPrice
+
+		// if txOpts.GasPrice wasn't previously set use the fetched gasPrice
+		// this should happen on the first iteration only
+		var previousTxGasPrice *big.Int
+		if txOpts.GasPrice == nil {
+			previousTxGasPrice = gasPrice
+		} else {
+			previousTxGasPrice = gasPrice
```

Comment:
I believe in this case you meant `txOpts.GasPrice`?

Reply:
Oh yes thanks!